src/BjyAuthorize/Service/CacheFactory.php

```diff
@@ -28,8 +28,21 @@ class CacheFactory implements FactoryInterface
      */
     public function createService(ServiceLocatorInterface $serviceLocator)
     {
+        $adapterName = null;
         $options = $serviceLocator->get('BjyAuthorize\Config');
+        if (is_array($options)
+            && isset($options['cache_options'])
+            && isset($options['cache_options']['adapter'])
+            && isset($options['cache_options']['adapter']['name'])
+        ) {
+            $adapterName = $options['cache_options']['adapter']['name'];
+        }
 
-        return StorageFactory::factory($options['cache_options']);
+        // Create adapter via serviceLocator if possible
+        if (!empty($adapterName) && $serviceLocator->has($adapterName)) {
```

Comment:
This just delegates to a cache service name. While it may appear as useful, I don't think it actually is. Replacing service [`'BjyAuthorize\Cache'`](https://github.com/bjyoungblood/BjyAuthorize/blob/87e10dac52506fe36fa8676292d46a3dce92099d/config/module.config.php#L58) is a much more effective way of achieving this, and without the layer of indirection either